src/ORM/ListDecorator.php

```diff
@@ -10,22 +10,34 @@
  * A base class for decorators that wrap around a list to provide additional
  * functionality. It passes through list methods to the underlying list
  * implementation.
+ *
+ * @template TList of SS_List&Sortable&Filterable&Limitable
+ * @template T
+ * @implements SS_List<T>
+ * @implements Sortable<T>
+ * @implements Filterable<T>
+ * @implements Limitable<T>
  */
 abstract class ListDecorator extends ViewableData implements SS_List, Sortable, Filterable, Limitable
 {
-
     /**
-     * @var SS_List
+     * @var TList<T>
```

Comment:
```suggestion
     * @var TList<T>|SS_List&Sortable&Filterable&Limitable
```
This is to resolve vscode complaining about strongly type `getList()` returning a `TList<T>` instead of a `SS_List&Sortable&Filterable&Limitable`

Reply:
`T` _is_ `SS_List&Sortable&Filterable&Limitable`.

I assume the complaint you're talking about is this red line here?
![image](https://github.com/silverstripe/silverstripe-framework/assets/36352093/26529f90-12ff-4590-9926-cc38761cf4a6)

If so, I'm inclined to ignore that. We know it's correct. When you call `getList()` on a list and look at what it's resolving to, it's clear vscode knows it's correct, too. Try this out and see what the type of `$y` is.
```php
/** @var ListDecorator $x */
$y = $x->getList();
```

This looks more like a bug with vscode or with intellephense to me, and I don't think we should cater our PHPDoc to bugs in IDE plugins.

Reply:
That said, I will make the change if you insist

Reply:
Yeah you're probably right, just omit my recommendation